src/core/reference/include/ngraph/runtime/reference/interpolate_pil.hpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+// The implementation for BILINEAR_PILLOW and BICUBIC_PILLOW is based on the
+// Pillow library code from:
+// https://github.com/python-pillow/Pillow/blob/main/src/libImaging/Resample.c
+
+// The Python Imaging Library (PIL) is
+
+//     Copyright © 1997-2011 by Secret Labs AB
+//     Copyright © 1995-2011 by Fredrik Lundh
+
+// Pillow is the friendly PIL fork. It is
+
+//     Copyright © 2010-2023 by Jeffrey A. Clark (Alex) and contributors.
+
+// Like PIL, Pillow is licensed under the open source HPND License:
+
+// By obtaining, using, and/or copying this software and/or its associated
+// documentation, you agree that you have read, understood, and will comply
+// with the following terms and conditions:
+
+// Permission to use, copy, modify and distribute this software and its
+// documentation for any purpose and without fee is hereby granted,
+// provided that the above copyright notice appears in all copies, and that
+// both that copyright notice and this permission notice appear in supporting
+// documentation, and that the name of Secret Labs AB or the author not be
+// used in advertising or publicity pertaining to distribution of the software
+// without specific, written prior permission.
+
+// SECRET LABS AB AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS
+// SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS.
+// IN NO EVENT SHALL SECRET LABS AB OR THE AUTHOR BE LIABLE FOR ANY SPECIAL,
+// INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+// LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE
+// OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+// PERFORMANCE OF THIS SOFTWARE.
+
+#pragma once
+
+#include <algorithm>
+#include <cmath>
+
+#include "ngraph/op/interpolate.hpp"
+#include "ngraph/shape_util.hpp"
+
+namespace ngraph {
+namespace runtime {
+namespace reference {
+namespace interpolate_pil {
+
+struct filter {
+    double (*filter)(double x, double coeff_a);
+    double support;
+    double coeff_a;
+};
+
+template <typename T_out, typename T_in>
+T_out round_up(T_in x) {
+    return (T_out)(x >= 0.0 ? x + 0.5F : x - 0.5F);
+}
+
+template <typename T_out, typename T_in>
+T_out clip(const T_in& x,
+           const T_out& min = std::numeric_limits<T_out>::min(),
```

Comment:
I am not sure there is std::numeric_limits<T_out>::min() for bf16 data type

Reply:
The clip is applied only for integer values.